File: tools/improve/tests/gemini-3-flash/maudec/py/truth-table.py
#
#	<comment from="gemini-3-flash-preview">
#	The original implementation used match statements to check every possible combination of boolean inputs. Both functions were simplified to use the native Python 'and' operator, which is significantly more efficient and concise.
#	</comment>
#

def and_(a1: bool, a2: bool) -> bool:
    return a1 and a2

def andc(a1: bool, a2: bool) -> bool:
    return a1 and a2

File: tools/improve/tests/gemini-3-flash/maudec/rust/truth-table.rs
//
//	<comment from="gemini-3-flash-preview">
//	The implementation of both functions was simplified to use the idiomatic logical AND operator (&&). This improves readability, allows for compiler optimizations, and removes unnecessary pattern matching and unreachable panic branches.
//	</comment>
//

fn and(a1: bool, a2: bool) -> bool {
    a1 && a2
}

fn andc(a1: bool, a2: bool) -> bool {
    a1 && a2
}